Remove ginkgo code

joshuarubin opened the pull request:
This simply removes references to ginkgo and gomega and replaces them with equivalent functionality from the standard Go testing framework.

**Which issue(s) this PR fixes**:
Fixes #545

**Release note**:
```release-note
NONE
```

joshuarubin commented:
Signed CLA

roberthbailey commented:
/ok-to-test

roberthbailey commented:
@joshuarubin - thanks for the PR! Overall it looks great. I'm going to put in a few stylistic comments, but they are all pretty minor.

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -76,34 +74,38 @@ func getNodeWithReadyStatus(nodeName string, nodeReadyStatus v1.ConditionStatus)
 
 	_, err = getClusterObject(c, "test-cluster", "get-cluster-object-with-no-cluster")
-	g.Expect(err).To(gomega.HaveOccurred())
+	if err == nil {
```
Checking err can be "cuddled" with the call in places where the call doesn't return anything that needs to be used outside of the conditional check:

`if _, err := getClusterObject(...); err == nil { ... }`

joshuarubin replied:
I like cuddling too! I often go to a new line when the present one is already pretty long, but I can easily make this change.

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -76,34 +74,38 @@ func getNodeWithReadyStatus(nodeName string, nodeReadyStatus v1.ConditionStatus)
 	cluster.Namespace = testNamespace
 	err = c.Create(context.TODO(), &cluster)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
Cases like this should be cuddled

`if err := c.Create(...); err != nil { ... }`

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -169,15 +171,19 @@ func TestGetClusterObjectWithMoreThanOneCluster(t *testing.T) {
 	cluster1.Namespace = testNamespace
 	err = c.Create(context.TODO(), &cluster1)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
cuddle here too

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -169,15 +171,19 @@ func TestGetClusterObjectWithMoreThanOneCluster(t *testing.T) {
 	cluster2.Namespace = testNamespace
 	err = c.Create(context.TODO(), &cluster2)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -263,18 +269,20 @@ func TestValidateClusterObject(t *testing.T) {
 	testNode := getNodeWithReadyStatus(testNodeName, v1.ConditionTrue)
 	err = c.Create(context.TODO(), &testNode)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -344,24 +352,28 @@ func TestValidateMachineObjects(t *testing.T) {
 	testNodeReady := getNodeWithReadyStatus(testNodeReadyName, v1.ConditionTrue)
 	err = c.Create(context.TODO(), &testNodeReady)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -344,24 +352,28 @@ func TestValidateMachineObjects(t *testing.T) {
 	testNodeNotReady := getNodeWithReadyStatus(testNodeNotReadyName, v1.ConditionFalse)
 	err = c.Create(context.TODO(), &testNodeNotReady)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			cluster.Namespace = testcase.namespace
 			err = c.Create(context.TODO(), &cluster)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
here too

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			machine1 := getMachineWithError(testMachine1Name, testcase.namespace, nil, nil, nil) // machine with no error
 			err = c.Create(context.TODO(), &machine1)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
here too

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			machine2 := getMachineWithError(testMachine2Name, testcase.namespace, nil, nil, nil) // machine with no error
 			err = c.Create(context.TODO(), &machine2)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			testNode1 := getNodeWithReadyStatus(testNode1Name, v1.ConditionTrue)
 			err = c.Create(context.TODO(), &testNode1)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			testNode2 := getNodeWithReadyStatus(testNode2Name, v1.ConditionTrue)
 			err = c.Create(context.TODO(), &testNode2)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
and here

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -483,32 +497,44 @@ func TestValidateClusterAPIObjectsOutput(t *testing.T) {
 			testNodeNotReady := getNodeWithReadyStatus(testNodeNotReadyName, v1.ConditionFalse)
 			err = c.Create(context.TODO(), &testNodeNotReady)
-			g.Expect(err).NotTo(gomega.HaveOccurred())
+			if err != nil {
```
and here

roberthbailey commented on pkg/apis/cluster/v1alpha1/cluster_types_test.go:
```diff
@@ -36,24 +36,39 @@ func TestStorageCluster(t *testing.T) {
 	// Test Create
 	fetched := &Cluster{}
-	g.Expect(c.Create(context.TODO(), created)).NotTo(gomega.HaveOccurred())
+	if err := c.Create(context.TODO(), created); err != nil {
```
This file was all done correctly. :)

roberthbailey commented on pkg/controller/cluster/cluster_controller_test.go:
```diff
@@ -51,18 +49,30 @@ func TestReconcile(t *testing.T) {
 	recFn, requests := SetupTestReconcile(newReconciler(mgr, a))
-	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
-	defer close(StartTestManager(mgr, g))
+	if err = add(mgr, recFn); err != nil {
```
`err :=`

roberthbailey commented on pkg/apis/cluster/v1alpha1/cluster_types_test.go:
```diff
@@ -36,24 +36,39 @@ func TestStorageCluster(t *testing.T) {
 	fetched := &Cluster{}
-	g.Expect(c.Create(context.TODO(), created)).NotTo(gomega.HaveOccurred())
+	if err := c.Create(context.TODO(), created); err != nil {
+		t.Errorf("error creating cluster: %v", err)
```
I didn't mention it earlier, but strings passed to `t.Errorf` or `t.Fatalf` should start with a capital letter, as they are line-oriented strings (as opposed to error strings which start with a lower case letter). Can you please fix across the PR?

roberthbailey commented on pkg/controller/machine/machine_controller_test.go:
```diff
@@ -48,19 +46,32 @@ func TestReconcile(t *testing.T) {
 	recFn, requests := SetupTestReconcile(newReconciler(mgr, a))
-	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
-	defer close(StartTestManager(mgr, g))
+	if err = add(mgr, recFn); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
 	recFn, requests, errors := SetupTestReconcile(r)
-	g.Expect(add(mgr, recFn, r.MachineSetToDeployments)).NotTo(gomega.HaveOccurred())
-	defer close(StartTestManager(mgr, g))
+	if err = add(mgr, recFn, r.MachineSetToDeployments); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
 	// Create the MachineDeployment object and expect Reconcile to be called.
-	err = c.Create(context.TODO(), instance)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err = c.Create(context.TODO(), instance); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
-	g.Eventually(func() int {
-		if err := c.List(context.TODO(), &client.ListOptions{}, machineSets); err != nil {
+	expectInt(t, 1, func(ctx context.Context) int {
+		if err = c.List(ctx, &client.ListOptions{}, machineSets); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
-	g.Eventually(errors, timeout).Should(gomega.Receive(gomega.BeNil()))
-	g.Eventually(func() int {
-		if err := c.List(context.TODO(), &client.ListOptions{}, machineSets); err != nil {
+	if err = c.Delete(context.TODO(), &ms); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
+	if err != nil {
+		t.Errorf("error scaling machinedeployment: %v", err)
+	}
+	if err = c.Update(context.TODO(), instance); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -69,72 +68,88 @@ func TestReconcile(t *testing.T) {
+	if err != nil {
+		t.Errorf("error scaling machinedeployment: %v", err)
+	}
+	if err = c.Update(context.TODO(), instance); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -150,50 +165,56 @@ func TestReconcile(t *testing.T) {
-	g.Expect(c.Status().Update(context.TODO(), oldMachineSet)).NotTo(gomega.HaveOccurred())
-	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
-	g.Eventually(errors, timeout).Should(gomega.Receive(gomega.BeNil()))
+	if err = c.Status().Update(context.TODO(), oldMachineSet); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -150,50 +165,56 @@ func TestReconcile(t *testing.T) {
-		g.Expect(c.Status().Update(context.TODO(), newMachineSet)).NotTo(gomega.HaveOccurred())
-		g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
-		g.Eventually(errors, timeout).Should(gomega.Receive(gomega.BeNil()))
+		if err = c.Status().Update(context.TODO(), newMachineSet); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machineset/machineset_controller_test.go:
```diff
@@ -54,37 +52,60 @@ func TestReconcile(t *testing.T) {
 	recFn, requests := SetupTestReconcile(r)
-	g.Expect(add(mgr, recFn, r.MachineSetToMachines)).NotTo(gomega.HaveOccurred())
-	defer close(StartTestManager(mgr, g))
+	if err = add(mgr, recFn, r.MachineSetToMachines); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machineset/machineset_controller_test.go:
```diff
@@ -54,37 +52,60 @@ func TestReconcile(t *testing.T) {
 	// Create the MachineSet object and expect Reconcile to be called and the Machines to be created.
-	err = c.Create(context.TODO(), instance)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+	if err = c.Create(context.TODO(), instance); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/machineset/machineset_controller_test.go:
```diff
@@ -54,37 +52,60 @@ func TestReconcile(t *testing.T) {
 	m := machines.Items[0]
-	g.Expect(c.Delete(context.TODO(), &m)).NotTo(gomega.HaveOccurred())
-	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
+	if err = c.Delete(context.TODO(), &m); err != nil {
```
`err :=`

roberthbailey commented on pkg/controller/node/node_controller_test.go:
```diff
@@ -38,18 +37,21 @@ var expectedRequest = reconcile.Request{NamespacedName: types.NamespacedName{Nam
 	recFn, requests := SetupTestReconcile(newReconciler(mgr))
-	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
-	defer close(StartTestManager(mgr, g))
+	if err = add(mgr, recFn); err != nil {
```
`err :=`

joshuarubin replied:
`err` is already defined above. By redefining it just for the scope of the `if` statement, we would be producing a "shadowed" variable which linting with `go tool vet -shadow` would (did) complain about. I'm new to the project and am not yet familiar with the coding standards. I can certainly make this change, but I just want to be clear about why I did it. LMK

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
+		for range time.Tick(pollingInterval) {
+			select {
+			case recv := <-requests:
+				if recv == expectedRequest {
```
where is `expectedRequest` defined?

joshuarubin replied:
It was already defined at the top of the file.

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
+			select {
+			case recv := <-requests:
+				if recv == expectedRequest {
+					break LOOP0
```
If you find what you wanted on the channel, why restart the loop?

joshuarubin replied:
I'm breaking from the loop, not restarting.

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
+			select {
+			case err := <-errors:
+				if err == nil {
+					break LOOP1
```
same question about this break; why start over instead of exiting?

joshuarubin replied:
The second loop (`LOOP1`) could just return, but I kept the semantics the same as the first loop for consistency. This will change when I do each loop concurrently.

roberthbailey commented on pkg/controller/machinedeployment/machinedeployment_controller_test.go:
```diff
@@ -205,3 +226,65 @@ func intstrPtr(i int32) *intstr.IntOrString {
+		ctx, cancel := context.WithTimeout(context.TODO(), timeout)
+		defer cancel()
+
+	LOOP0:
```
Should loop0 and loop1 be done in parallel instead of sequentially?

joshuarubin replied:
Sure, I like that idea.

joshuarubin commented:
OK, I've addressed the issues brought up in review (except for the `err :=` issue which I'm waiting for some clarification on).

joshuarubin commented:
/retest

joshuarubin commented:
Huh, this is definitely a flake, looking into it...

roberthbailey commented:
A couple of examples are shown in https://golang.org/doc/effective_go.html#if & https://golang.org/doc/effective_go.html#blank_assign.

joshuarubin commented:
Like I said, I don’t mind doing it.

Did you try running the command exactly as stated? `go tool vet -shadow`? You can see the help with `go tool vet -h`, not the same as just `go vet`. It's obviously "experimental":

```
...
  -shadow
        check for shadowed variables (experimental; must be set explicitly)
  -shadowstrict
        whether to be strict about shadowing; can be noisy
...
```

The issue isn't so much creating the variable for the nearest scope, just that it conflicts with the same variable name in the outer scope.

Seriously though, I'll just update it. It's not a big deal.

joshuarubin commented:
OK, updated.

roberthbailey commented on cmd/clusterctl/validation/validate_cluster_api_objects_test.go:
```diff
@@ -76,34 +74,36 @@ func getNodeWithReadyStatus(nodeName string, nodeReadyStatus v1.ConditionStatus)
 
-	_, err = getClusterObject(c, "test-cluster", "get-cluster-object-with-no-cluster")
-	g.Expect(err).To(gomega.HaveOccurred())
+	if _, err = getClusterObject(c, "test-cluster", "get-cluster-object-with-no-cluster"); err == nil {
```
This is the only one left....

joshuarubin commented:
> Thanks! It looks like you just missed one....

Bah, I fixed that, then forced an update, but the PR wasn't updated... anyway, my git foo failed me. Updated now.

roberthbailey commented:
/lgtm
/approve
